dockerfiles/Dockerfile.training

```
@@ -0,0 +1,193 @@
+    python setup.py bdist_wheel -d build/wheel &&\
+    pip install build/wheel/*.whl
+
+# build onnxruntime wheel with cuda and mpi support
+RUN cd /stage && git clone https://github.com/microsoft/onnxruntime.git &&\
+    cd onnxruntime &&\
+    git checkout v1.4.0 &&\
+    python tools/ci_build/build.py \
+        --cmake_extra_defines \
+            ONNXRUNTIME_VERSION=`cat ./VERSION_NUMBER` \
+        --config RelWithDebInfo \
+        --enable_training \
+        --mpi_home /opt/ompi4 \
+        --use_cuda \
+        --cuda_home /usr/local/cuda \
+        --cudnn_home /usr/lib/x86_64-linux-gnu/ \
+        --nccl_home /usr/local/cuda/targets/x86_64-linux \
+        --update \
+        --parallel \
+        --build_dir build \
+        --build \
+        --build_wheel \
+        --skip_tests &&\
+    pip install build/RelWithDebInfo/dist/*.whl
+
+# switch to cuda runtime environment
+# note: launch with --gpus all or nvidia-docker
+FROM nvidia/cuda:10.2-cudnn7-runtime-ubuntu18.04
+RUN apt-get -y update
+WORKDIR /stage
+
+# install mellanox libraries (again)
+# note: launch with infiniband drivers and --net=host or docker swarm
+ARG MOFED_VERSION
+COPY --from=build /stage/mofed-${MOFED_VERSION} /stage/mofed-${MOFED_VERSION}
+RUN apt-get install -y build-essential &&\
+    cd /stage/mofed-${MOFED_VERSION} &&\
+    ./mlnxofedinstall --basic --force --user-space-only --without-fw-update
+
+# install ucx
+# note: launch with --cap-add=sys_nice to avoid 'mbind' warnings
+COPY --from=build /opt/ucx /opt/ucx
+ENV PATH=/opt/ucx/bin:$PATH
+ENV LD_LIBRARY_PATH=/opt/ucx/lib:$LD_LIBRARY_PATH
+
+# install openmpi
+# note: permit mpirun as root for Azure cluster submissions
+# note: enforce openmpi select ucx or fail
+COPY --from=build /opt/ompi4 /opt/ompi4
+ENV PATH=/opt/ompi4/bin:$PATH
+ENV LD_LIBRARY_PATH=/opt/ompi4/lib:$LD_LIBRARY_PATH
+ENV OMPI_ALLOW_RUN_AS_ROOT=1
+ENV OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
+ENV OMPI_MCA_pml=ucx
+RUN apt-get install -y \
+        openssh-server \
+        openssh-client \
+        libnuma-dev &&\
+    ldconfig
+
+# copy conda environment (includes mkl, numpy, pytorch, onnxruntime)
+COPY --from=build /opt/conda /opt/conda
+ENV PATH=/opt/conda/bin:${PATH}
+
+# make ssh/sshd less strict for wiring containers on Azure VM scale set
+# note: use 'service ssh start' to launch sshd (will fail if 22 in use)
+# note: can also set port != 22 and set port=X in MPI hosts file
+# note: need to setup password free ssh login between MPI hosts
+RUN sed -i 's/#PermitRootLogin prohibit-password/PermitRootLogin yes/g' \
+        /etc/ssh/sshd_config &&\
+    sed -i 's/#StrictModes yes/StrictModes no/g' \
+        /etc/ssh/sshd_config &&\
+    sed -i 's/#   StrictHostKeyChecking ask/    StrictHostKeyChecking no/g' \
+        /etc/ssh/ssh_config &&\
+    mkdir /run/sshd &&\
+    chmod u=rwx,go=rx /run/sshd
+
+# clean\finalize environment
+RUN conda remove -y cmake ninja &&\
+    apt-get remove -y build-essential &&\
```

```suggestion
    apt-get purge -y build-essential &&\
```
this uninstall and delete old configuration files